### ISSUE TITLE
get rid of the DB

### ISSUE THREAD

**Issue by maxdemaio** [maxdemaio/thelangbot#31]
We want to get rid of the DB. Creates a point of failure and we don't even really need one. TBH we don't even store sensitive data. We could just legit have text files. We will just update them in version control. All we have are blacklisted/banner users (ban by immutable IDs), supporter IDs (people that have supported the bot and get likes alongside retweets), and then we'll try and remove storing the "last seen ID" because of #28.

**Comment by Rairye** [comment]
It is actually possible to manage this without using serialization.

**Comment by maxdemaio** [comment]
I was thinking sqlite and we could actually commit the files because nothing is really sensitive information

**Comment by maxdemaio** [comment]
Ideally we should only use the DB for banned/blacklisted users. Honestly wouldn't even need to be a DB maybe even a text file.

**Comment by maxdemaio** [comment]
Advice from @Rairye:

1. Grab the most recent tweet from thelangbot timeline (I think you can filter by retweets, so as to avoid tweets that you manually made)

2. If the difference between the current time and timestamp of the tweet is longer than certain period of time (like 30 minutes), grab the most recent x amount of tweets with the target hashtags and then break when the timestamp of the tweet is longer than a certain period of time (like, 15 minutes?)

3.  If the timestamp of the last retweet is within a certain period of time, use that tweet id to search for the next batch of tweets to retweet

**Comment by maxdemaio** [comment]
Currently the bot will go down in the situation where the app fails to do an UPDATE on the MySQL database. At the very end of retweeting tweets, the bot will attempt to store the last seen ID into the DB. If it fails, this causes the last seen tweet to be stale, and if the bot re-queries Twitter, most subsequent tweets from the stale ID will have already been retweeted.

Error seen from logs:

```
2022-04-28T13:22:41.643519+00:00 app[scheduler.3653]:     Utils.storeLastSeenId(mydb, mycursor, currLastSeenId)
2022-04-28T13:22:41.643519+00:00 app[scheduler.3653]:   File "/app/utils.py", line 41, in storeLastSeenId
2022-04-28T13:22:41.643580+00:00 app[scheduler.3653]:     mycursor.execute("UPDATE tweet SET tweetId = '%s' WHERE id = 1", (exampleId,))
2022-04-28T13:22:41.643582+00:00 app[scheduler.3653]:   File "/app/.heroku/python/lib/python3.9/site-packages/mysql/connector/cursor_cext.py", line 269, in execute
2022-04-28T13:22:41.643692+00:00 app[scheduler.3653]:     result = self._cnx.cmd_query(stmt, raw=self._raw,
2022-04-28T13:22:41.643693+00:00 app[scheduler.3653]:   File "/app/.heroku/python/lib/python3.9/site-packages/mysql/connector/connection_cext.py", line 510, in cmd_query
2022-04-28T13:22:41.643837+00:00 app[scheduler.3653]:     raise errors.get_mysql_exception(exc.errno, msg=exc.msg,
2022-04-28T13:22:41.643856+00:00 app[scheduler.3653]: mysql.connector.errors.OperationalError: 2013 (HY000): Lost connection to MySQL server during query
```

**Comment by maxdemaio** [comment]
closed with #34
